lib/Backend/GlobOpt.cpp

```diff
@@ -3825,32 +3826,157 @@ GlobOpt::TestAnyArgumentsSym()
+{
+    IR::Opnd* dst = instr->GetDst();
+    IR::Opnd* src1 = instr->GetSrc1();
+
+    if (instr->m_opcode == Js::OpCode::Ld_A && src1->IsRegOpnd())
+    {
+        AssertMsg(!src1->IsScopeObjOpnd(instr->m_func), "There can be no aliasing for scope object.");
+    }
+
+    // The following is to track formals array for Stack Arguments optimization with Formals
+    if (instr->IsStackArgsEnabled() && !this->IsLoopPrePass())
+    {
+        if (instr->m_opcode == Js::OpCode::LdSlotArr)
+        {
+            if (instr->GetSrc1()->IsScopeObjOpnd(instr->m_func))
+            {
+                AssertMsg(!instr->m_func->GetJnFunction()->GetHasImplicitArgIns(), "No mapping is required in this case. So it should already be generating ArgIns.");
+                instr->m_func->TrackFormalsArraySym(dst->GetStackSym()->m_id);
+            }
+        }
+        else if (instr->m_opcode == Js::OpCode::InlineeStart)
+        {
+            Assert(instr->m_func->IsInlined());
+            Js::ArgSlot actualsCount = instr->m_func->actualCount - 1;
+            Js::ArgSlot formalsCount = instr->m_func->GetJnFunction()->GetInParamsCount() - 1;
+
+            Func * func = instr->m_func;
+
+            IR::Instr * argOutInstr = instr->GetSrc2()->GetStackSym()->GetInstrDef();
+
+            //The argout immediately before the InlineeStart will be the ArgOut for NewScObject
+            //So we don't want to track the stack sym for this argout.- Skipping it here.
+            if (instr->m_func->IsInlinedConstructor())
+            {
+                Assert(argOutInstr->GetSrc1()->GetStackSym()->GetInstrDef() != nullptr);
+                Assert(argOutInstr->GetSrc1()->GetStackSym()->GetInstrDef()->m_opcode == Js::OpCode::NewScObjectNoCtor);
+                argOutInstr = argOutInstr->GetSrc2()->GetStackSym()->GetInstrDef();
+            }
+            if (formalsCount < actualsCount)
+            {
+                Js::ArgSlot extraActuals = actualsCount - formalsCount;
+
+                //Skipping extra actuals passed
+                for (Js::ArgSlot i = 0; i < extraActuals; i++)
+                {
+                    argOutInstr = argOutInstr->GetSrc2()->GetStackSym()->GetInstrDef();
+                }
+            }
+
+            StackSym * undefinedSym = nullptr;
+
+            for (Js::ArgSlot param = formalsCount; param > 0; param--)
+            {
+                StackSym * argOutSym = nullptr;
+
+                if (argOutInstr->GetSrc1())
+                {
+                    if (argOutInstr->GetSrc1()->IsRegOpnd())
+                    {
+                        argOutSym = argOutInstr->GetSrc1()->GetStackSym();
+                    }
+                    else
+                    {
+                        //TODO: saravind: Can I just use the same source operand like this?
+                        // We will always have ArgOut instr - so the source operand will not be removed.
+                        argOutSym = StackSym::New(func);
```

Comment:
Fixed

---
In reply to: [67799824](https://github.com/Microsoft/ChakraCore/pull/1135#discussion_r67799824) [](ancestors = 67799824,67736248)